### PR TITLE
Hide ads when doc becomes inactive and show them again.

### PR DESCRIPTION
**Pull request by cramforce**
This should solve major perf problems with ads in swiped away docs and reduce CPU/GPU/Battery consumption. A more robust solution is being developed by @jridgewell.

@dvoytenko Looking good?

### CODE REVIEW

File: builtins/amp-ad.js [hunk 1]
```diff
@@ -203,24 +208,29 @@ export function installAd(win) {
...
-        // now.
-        loadingAdsCount--;
-      }, 1000);
-      assert(!this.isInFixedContainer_,
-          '<amp-ad> is not allowed to be placed in elements with ' +
-          'position:fixed: %s', this.element);
-      this.element.setAttribute('scrolling', 'no');
+      if (this.paused_) {
+        this.paused_ = false;
+        this.iframe_.style.display = '';
```
Comment (dvoytenko) [inline builtins/amp-ad.js]: There's style.js `toggle` method.

File: builtins/amp-ad.js [hunk 2]
```diff
@@ -257,6 +267,18 @@ export function installAd(win) {
       return loadPromise(this.iframe_);
     }
 
+    /** @override */
+    documentInactiveCallback() {
+      if (this.iframe_) {
+        this.paused_ = true;
+        // When the doc is inactive, hide the ads, so any work they do takes
+        // less CPU and power.
+        this.iframe_.style.display = 'none';
```
Comment (dvoytenko) [inline builtins/amp-ad.js]: ditto on toggle

**Comment by dvoytenko** [comment]
@cramforce Yeah, looks good.

File: builtins/amp-ad.js (hunk 2 above)
Comment (erwinmombay) [inline builtins/amp-ad.js]: qq, should we be displaying none or doing visiblity hidden?

Comment (cramforce) [inline builtins/amp-ad.js]: `Display:none`. Visibility has all of the cost with the downside of the ad not showing.

File: builtins/amp-ad.js [hunk 3]
```diff
@@ -106,6 +108,9 @@ export function installAd(win) {
 
       /** @private {IntersectionObserver} */
       this.intersectionObserver_ = null;
+
+      /** @private {boolean}  */
+      this.paused_ = false;
```
Comment (sriramkrish85) [inline builtins/amp-ad.js]: could you annotate what paused means here?

File: builtins/amp-ad.js (hunk 2 above)
Comment (cramforce) [inline builtins/amp-ad.js]: Done

File: builtins/amp-ad.js (hunk 1 above)
Comment (cramforce) [inline builtins/amp-ad.js]: Done.

File: builtins/amp-ad.js (hunk 2 above)
Comment (cramforce) [inline builtins/amp-ad.js]: Done

**Comment by cramforce** [comment]
Added tests

**Pull request merged**